src/net_processing.cpp

```diff
@@ -362,10 +362,9 @@ class PeerManagerImpl final : public PeerManager
 
     /** Number of outbound peers with m_chain_sync.m_protect. */
     int m_outbound_peers_with_protect_from_disconnect GUARDED_BY(cs_main) = 0;
-};
-} // namespace
 
-namespace {
+    bool AlreadyHaveTx(const GenTxid& gtxid, const CTxMemPool& mempool) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
eeac506

I don't know if we have code guidelines on this but maybe move this method at the bottom of the other ones instead of among variables members. 

Also maybe add a small comment "Check if we have already seen this transaction by looking up its identifier among reject filter, orphan pool, recent confirmations and mempool" ?

Reply:
See discussion here: https://github.com/bitcoin/bitcoin/pull/20942#discussion_r559495985